torch/jit/__init__.py

```diff
@@ -463,149 +463,14 @@ def wrapper(func):
         for name in executor_options:
             executor_options[name] = kwargs.pop(name, executor_options[name])
         if isinstance(func, torch.nn.Module):
-            captures = list(func.state_dict(keep_vars=True).values())
-            # TODO: support shared parameters
-            if len(set(map(id, captures))) != len(list(map(id, captures))):
-                raise ValueError("TracedModules don't support parameter sharing between modules")
-            executor = torch._C.GraphExecutor(func, args, captures=captures, **executor_options)
-            return TracedModule(func, executor)
+            module = TracedModule(func, **executor_options)
+            module._create_method_from_trace('forward', func, args)
+            return module
```

Comment:
Have you benchmarked this change?

Reply:
No, it would be a good idea to do, but I want to unblock @jamesr66a first. I think conceptually the timing will not change just because we load parameters via pointers, but it is of course possible this adds some other overhead, so timing is a good idea.